core/capabilities/compute/compute_test.go

```diff
@@ -132,7 +171,87 @@ func Test_Compute_Execute(t *testing.T) {
 			ReferenceID: "compute",
 		},
 	}
-	resp, err = compute.Execute(tests.Context(t), req)
+	resp, err = th.compute.Execute(tests.Context(t), req)
 	assert.NoError(t, err)
 	assert.False(t, resp.Value.Underlying["Value"].(*values.Bool).Underlying)
 }
+
+func TestComputeFetch(t *testing.T) {
+	workflowID := "15c631d295ef5e32deb99a10ee6804bc4af13855687559d7ff6552ac6dbb2ce0"
+	workflowExecutionID := "95ef5e32deb99a10ee6804bc4af13855687559d7ff6552ac6dbb2ce0abbadeed"
+	th := setup(t, defaultConfig)
+
+	th.connector.EXPECT().DonID().Return("don-id")
+	th.connector.EXPECT().GatewayIDs().Return([]string{"gateway1", "gateway2"})
+
+	msgID := strings.Join([]string{
+		workflowID,
+		workflowExecutionID,
+		ghcapabilities.MethodComputeAction,
+		validRequestUUID,
+	}, "/")
+
+	gatewayResp := gatewayResponse(t, msgID)
+	th.connector.On("SignAndSendToGateway", mock.Anything, "gateway1", mock.Anything).Return(nil).Run(func(args mock.Arguments) {
+		th.connectorHandler.HandleGatewayMessage(context.Background(), "gateway1", gatewayResp)
+	}).Once()
+
+	require.NoError(t, th.compute.Start(tests.Context(t)))
+
+	binary := wasmtest.CreateTestBinary(fetchBinaryCmd, fetchBinaryLocation, true, t)
+
+	config, err := values.WrapMap(map[string]any{
+		"config": []byte(""),
+		"binary": binary,
+	})
+	require.NoError(t, err)
+
+	req := cappkg.CapabilityRequest{
+		Config: config,
+		Metadata: cappkg.RequestMetadata{
+			WorkflowID:          workflowID,
+			WorkflowExecutionID: workflowExecutionID,
+			ReferenceID:         "compute",
+		},
+	}
+
+	headers, err := values.NewMap(map[string]any{})
+	require.NoError(t, err)
+	expected := cappkg.CapabilityResponse{
+		Value: &values.Map{
+			Underlying: map[string]values.Value{
+				"Value": &values.Map{
+					Underlying: map[string]values.Value{
+						"Body":       values.NewBytes([]byte("response body")),
+						"Headers":    headers,
+						"StatusCode": values.NewInt64(200),
+						"Success":    values.NewBool(true),
+					},
+				},
+			},
+		},
+	}
+
+	actual, err := th.compute.Execute(tests.Context(t), req)
+	assert.NoError(t, err)
+	assert.EqualValues(t, expected, actual)
+}
+
+func gatewayResponse(t *testing.T, msgID string) *api.Message {
+	headers := map[string]any{"Content-Type": "application/json"}
+	body := []byte("response body")
+	responsePayload, err := json.Marshal(sdk.FetchResponse{
```

Comment:
This should be the generic struct that we use, i.e. TargetResponsePayload (as I highlight though, we should change the name so it's more generic)